### PR TITLE
Implement `RUF010` to detect explicit type conversions within f-strings

### PR DESCRIPTION
**Pull request by LotemAm**
This rule will check if an f-string contains anything that can be replace with a conversion (e.g. `str(foo)` to `foo!s`)

Sorry if there are any glaring Rust mistakes, this is my first time with the language

### CODE REVIEW

File: crates/ruff/src/rules/ruff/rules/f_string_conversion.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            return;
+        }
+
+        let ExprKind::Name(ast::ExprName { id, .. }) = &func.node else {
+            return;
+        };
+
+        match id.as_str() {
+            "ascii" | "str" | "repr" => {
+                let mut diagnostic = Diagnostic::new(FStringConversion, formatted_value.range());
```
Comment (MichaReiser) [inline crates/ruff/src/rules/ruff/rules/f_string_conversion.rs:55]: I think we must check that the functions resolve to the `builtin` functions and not to user-defined functions.

Comment (LotemAm) [reply crates/ruff/src/rules/ruff/rules/f_string_conversion.rs:66]: Agreed, I added a check

Comment (MichaReiser) [reply crates/ruff/src/rules/ruff/rules/f_string_conversion.rs:66]: I think you can do `checker.ctx.is_builtin(id)` directly

Comment (LotemAm) [reply crates/ruff/src/rules/ruff/rules/f_string_conversion.rs:59]: Much better, thanks!

**Comment by charliermarsh** [comment]
Will merge this after cutting the release.

**Pull request merged**